velodyne_driver/src/driver/driver.cc

```diff
@@ -172,7 +185,7 @@ bool VelodyneDriver::poll(void)
 
   // publish message using time of last packet read
   ROS_DEBUG("Publishing a full Velodyne scan.");
-  scan->header.stamp = ros::Time(scan->packets[config_.npackets - 1].stamp);
+  scan->header.stamp = scan->packets[config_.npackets - 1].stamp;
```

Comment:
feels like this should be `scan->packets.back().stamp;`. A scan with cut_angle set can have less than `npackets` packets.

Reply:
You are exactly right. Thanks for finding this bug. The bug caused a duplication of timestamps. I will fix this immediately.